java-manta-it/src/test/java/com/joyent/manta/client/helper/IntegrationTestHelper.java

```diff
@@ -1,59 +1,116 @@
-        String bucketPathPrefix = String.format("%s%sbuckets%s%s", config.getMantaHomeDirectory(),
-                MantaClient.SEPARATOR, MantaClient.SEPARATOR,
-                testName.toLowerCase() + UUID.randomUUID());
-        return bucketPathPrefix + MantaClient.SEPARATOR + "objects" + MantaClient.SEPARATOR;
+        final String bucketPathPrefix = String.format("%s%s%s", config.getMantaBucketsDirectory(),
+                SEPARATOR, integrationTestBucket);
+        return bucketPathPrefix + bucketObjectsSubstring + testName + "-";
     }
 
-    private static boolean verifyBucketsSupport(final ConfigContext config,
+    /**
+     * Verify Buckets support in Manta.
+     * @see MantaClient#options(String)
+     */
+    public static boolean verifyBucketsSupport(final ConfigContext config,
                                                final MantaClient client) throws IOException {
         try {
-            String path = String.format("%s%sbuckets", config.getMantaHomeDirectory(),
-                    MantaClient.SEPARATOR);
+            String path = config.getMantaBucketsDirectory();
             client.options(path);
-        } catch (MantaClientHttpResponseException me) {
-            if (MantaErrorCode.RESOURCE_NOT_FOUND_ERROR.equals(me.getServerCode())) {
-                return false;
+        } catch (MantaClientHttpResponseException e) {
+            if (MantaErrorCode.RESOURCE_NOT_FOUND_ERROR.equals(e.getServerCode())) {
+                LOG.info("Buckets not supported in current Manta: {}", e.getStatusMessage());
             }
+            return false;
         }
         return true;
     }
 
+    /**
+     * Deletes all test data for integration-tests.
+     *
+     * @param client Manta Client instance for the operations.
+     * @param testPath The fully qualified test path for integration-tests.
+     * @throws MantaClientHttpResponseException test-bucket being used for integration-tests is not empty.
+     * @throws IOException If an IO exception has occurred.
+     */
     public static void cleanupTestBucketOrDirectory(final MantaClient client,
                                           final String testPath) throws IOException {
-
-        String bucketObjectsSubstring = String.format("%sobjects%s", MantaClient.SEPARATOR,
-                MantaClient.SEPARATOR);
         if (testPath.contains(bucketObjectsSubstring)) {
-            String bucketPath = testPath.substring(0, testPath.lastIndexOf(bucketObjectsSubstring));
-            //Delete bucket here
+            final String bucket_path = parseBucketPath(testPath);
+
+            try {
+                client.deleteBucket(bucket_path);
+            } catch (MantaClientHttpResponseException e) {
+                if (e.getServerCode().equals(MantaErrorCode.BUCKET_NOT_EMPTY_ERROR)) {
+                    e.setContextValue("Cleanup of non-empty test bucket attempted at:", testPath);
```

Comment:
You can set the context value as ```e.setContextValue("bucket_path:", testPath);``` and then use a message when throwing the exception something like ```throw new MantaIOException("Cannot delete bucket containing objects", e);```

Reply:
Noted.